model/plugin_kvset_options.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// PluginKVSetOptions contains information on how to store a value in the plugin KV store.
+type PluginKVSetOptions struct {
+	EncodeJSON      bool        // If true store the information encoded as JSON
+	Atomic          bool        // Only store the value if the current value matches the oldValue
```

Comment:
Maybe: (1/5)
```suggestion
	Atomic          bool        // Using compare-and-set, only store the value if the current value matches the oldValue
```

Reply:
I would like to hide how we are doing this. 3/5

Reply:
Agree with @hanzei on this one.